### PR TITLE
Added support for reviewers

### PR DESCRIPTION
**Pull request by Laurent-BH**

### CODE REVIEW

File: NGitLab/Models/MergeRequestQuery.cs [hunk 1]
```diff
@@ -74,6 +74,11 @@ public class MergeRequestQuery
         /// </summary>
         public QueryAssigneeId AssigneeId { get; set; }
 
+        /// <summary>
+        /// Returns the merge requests reviewer ids.
+        /// </summary>
+        public int[] ReviewerIds { get; set; }
```
Comment (Toa741) [inline NGitLab/Models/MergeRequestQuery.cs]: There's no reviewer_ids field for merge request query.

There's a reviewer_id field however

File: NGitLab/Impl/MergeRequestClient.cs [hunk 2]
```diff
@@ -46,6 +46,7 @@ public IEnumerable<MergeRequest> Get(MergeRequestQuery query)
             url = Utils.AddParameter(url, "author_id", query.AuthorId);
             url = Utils.AddParameter(url, "per_page", query.PerPage);
             url = Utils.AddParameter(url, "assignee_id", query.AssigneeId);
+            url = Utils.AddParameter(url, "reviewer_ids[]", query.ReviewerIds);
```
Comment (Toa741) [inline NGitLab/Impl/MergeRequestClient.cs]: As the comment below: reviewer_ids does not exists. There's only reviewer_id

File: NGitLab.Mock/Clients/MergeRequestClient.cs [hunk 3]
```diff
@@ -377,6 +377,12 @@ public IEnumerable<Models.MergeRequest> FilterByQuery(IEnumerable<MergeRequest>
                     mergeRequests = mergeRequests.Where(mr => mr.Assignee?.Id == assigneeId);
                 }
 
+                if (query.ReviewerIds != null)
+                {
+                    var reviewerIds = query.ReviewerIds;
+                    mergeRequests = mergeRequests.Where(mr => mr.Reviewers.Any(x => reviewerIds.Contains(x.Id)));
```
Comment (Toa741) [inline NGitLab.Mock/Clients/MergeRequestClient.cs]: Replace with reviewer_id and mock correctly with a "QueryReviewerId"

File: NGitLab.Mock/Clients/MergeRequestClient.cs [hunk 4]
```diff
@@ -571,6 +577,24 @@ public Models.MergeRequest Update(int mergeRequestIid, MergeRequestUpdate mergeR
...
+                if (mergeRequestUpdate.ReviewerIds != null)
+                {
+                    var users = new List<UserRef>();
+
+                    foreach (var userId in mergeRequestUpdate.ReviewerIds)
+                    {
+                        users.Add(Server.Users.GetById(userId));
+                    }
+
+                    if (!users.Any())
```
Comment (Toa741) [inline NGitLab.Mock/Clients/MergeRequestClient.cs]: put the check up to check on Server.Users.GetById(userId), update the error message with the not found user if that's the case.

No need to test on empty list

File: NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj [hunk 5]
```diff
@@ -1,4 +1,6 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project>
+  <Import Project="$([System.IO.Path]::Combine($(SolutionDir), '.SwitchToSource', $(SolutionFileName.Replace('.SwitchToSource.sln', '.sln')), 'import', 'NGitLab.Mock.Tests.before.props'))" Condition="$(SolutionFileName.EndsWith('.SwitchToSource.sln', StringComparison.OrdinalIgnoreCase)) And Exists($([System.IO.Path]::Combine($(SolutionDir), '.SwitchToSource', $(SolutionFileName.Replace('.SwitchToSource.sln', '.sln')), 'import', 'NGitLab.Mock.Tests.before.props')))" />
```
Comment (louis-z) [inline NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj]: This header and the footer below were modified by SwitchToSource, which is an internal tool (at least for now). Please revert.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab.Mock/MergeRequest.cs [hunk 6]
```diff
@@ -27,6 +27,10 @@ public MergeRequest()
 
         public UserRef Assignee { get; set; }
 
+        public UserRef Reviewer { get; set; }
```
Comment (Toa741) [inline NGitLab.Mock/MergeRequest.cs]: There's no reviewer on Get, only reviewers

File: NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj (hunk 5 above)
Comment (louis-z) [reply NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj]: Still unresolved

Comment (louis-z) [reply NGitLab.Mock.Tests/NGitLab.Mock.Tests.csproj]: The file should be reverted completely and look like this:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFrameworks>net5.0;net472</TargetFrameworks>
    <IsPackable>false</IsPackable>
  </PropertyGroup>
  <ItemGroup>
    <ProjectReference Include="..\NGitLab.Mock\NGitLab.Mock.csproj" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />
    <PackageReference Include="NUnit" Version="3.13.2" />
    <PackageReference Include="NUnit3TestAdapter" Version="4.0.0" />
  </ItemGroup>
</Project>

```

File: NGitLab.Mock/Clients/MergeRequestClient.cs [hunk 7]
```diff
@@ -377,6 +377,12 @@ public IEnumerable<Models.MergeRequest> FilterByQuery(IEnumerable<MergeRequest>
                     mergeRequests = mergeRequests.Where(mr => mr.Assignee?.Id == assigneeId);
                 }
 
+                if (query.ReviewerId != null)
+                {
+                    var reviewerId = string.Equals(query.ReviewerId.ToString(), "None", StringComparison.OrdinalIgnoreCase) ? (int?)null : int.Parse(query.ReviewerId.ToString());
```
Comment (Toa741) [inline NGitLab.Mock/Clients/MergeRequestClient.cs]: If Any => no need to filter
If None => should only take mr with no reviewer
If Id is set => should take mr if reviewer is present

**Review by louis-z — APPROVED** [review]

File: NGitLab.Mock/Clients/MergeRequestClient.cs (hunk 7 above)
Comment (Laurent-BH) [reply NGitLab.Mock/Clients/MergeRequestClient.cs]: I don't fully understand, what is Any or None here?

Comment (Toa741) [reply NGitLab.Mock/Clients/MergeRequestClient.cs]: Check the documentation: 
https://docs.gitlab.com/ee/api/merge_requests.html#list-merge-requests

```
Returns merge requests which have the user as a reviewer with the given user id. None returns merge requests with no reviewers. Any returns merge requests with any reviewer. Mutually exclusive with reviewer_username.
```
